app/src/main/java/org/oppia/app/drawer/ExitProfileDialogFragment.kt

```diff
@@ -53,7 +72,11 @@ class ExitProfileDialogFragment : DialogFragment() {
       .setMessage(R.string.home_activity_back_dialog_message)
       .setNegativeButton(R.string.home_activity_back_dialog_cancel) { dialog, _ ->
         if (isFromNavigationDrawer) {
-          exitProfileDialogInterface.markHomeMenuCloseDrawer()
+          exitProfileDialogInterface.markLastCheckedItemCloseDrawer(
```

Comment:
@anandwana001 So, to look into it deeper, I commented out the newly written methods. And when I tested it, I found out that now after canceling the dialog box, ```switch profile``` gets marked.